### PR TITLE
Adding node pool expansion for GKE cluster.

### PR DESCRIPTION
**Pull request by soumyapani**
Validated node pool expansion by creating a GKE cluster with 1 node_count and then changing the node_count to 2. It does not destroy the node pool simply changes the node count.

### CODE REVIEW

**Review by stevenBorisko — APPROVED** [review]

File: examples/gkeenv.list [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+NAME_PREFIX=test-aiinfra-gke
+ZONE=us-central1-a
+ACCELERATOR_TYPE=nvidia-tesla-a100
+METADATA={ meta1 = "val", meta2 = "val2" }
+LABELS={ label1 = "marker1" }
+ORCHESTRATOR_TYPE=gke
+INSTANCE_COUNT=1
+NETWORK_CONFIG=default_network
+GPU_COUNT=2
+GKE_NODE_POOL_COUNT=2
```
Comment (DmitryKakurin) [inline examples/gkeenv.list:11]: How do these 3 params below co-exist with each other? Does the last one takes precedence if present, and makes the first 2 unnecessary?

Comment (soumyapani) [reply examples/gkeenv.list:11]: Actually the contention is between `GKE_NODE_COUNT` and `INSTANCE_COUNT`. Everything else is needed whether GKE or not.
The `GKE_NODE_COUNT` takes effect only if `ORCHESTRATOR_TYPE=gke` and takes precedence over  `INSTANCE_COUNT`.

Comment (soumyapani) [reply examples/gkeenv.list:8]: renamed `GKE_NODE_COUNT` to `GKE_NODE_COUNT_PER_NODE_POOL` and added validations as discussed.

File: examples/gkeenv.list [hunk 2]
```diff
@@ -2,12 +2,9 @@ PROJECT_ID=testing-project-id
...
 ORCHESTRATOR_TYPE=gke
-INSTANCE_COUNT=1
 NETWORK_CONFIG=default_network
 GPU_COUNT=2
 GKE_NODE_POOL_COUNT=2
-GKE_NODE_COUNT=2
-CUSTOM_NODE_POOL=[{"name"="sp-test-pool-1","node_count"=2,"machine_type"="a2-highgpu-2g","guest_accelerator_count"=2,"guest_accelerator_type"="nvidia-tesla-a100"},{"name"="sp-test-pool-2","node_count"=2,"machine_type"="a2-highgpu-2g","guest_accelerator_count"=2,"guest_accelerator_type"="nvidia-tesla-a100"}]
\ No newline at end of file
+GKE_NODE_COUNT_PER_NODE_POOL=2
+# CUSTOM_NODE_POOL=[{"name"="sp-test-pool-1","node_count"=2,"machine_type"="a2-highgpu-2g","guest_accelerator_count"=2,"guest_accelerator_type"="nvidia-tesla-a100"},{"name"="sp-test-pool-2","node_count"=2,"machine_type"="a2-highgpu-2g","guest_accelerator_count"=2,"guest_accelerator_type"="nvidia-tesla-a100"}]
```
[the remark this answers is not in the document]
Comment (soumyapani) [reply examples/gkeenv.list:10]: sure. we can do that

File: examples/gkeenv.list [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+ZONE=us-central1-a
+ACCELERATOR_TYPE=nvidia-tesla-a100
+METADATA={ meta1 = "val", meta2 = "val2" }
+LABELS={ label1 = "marker1" }
+ORCHESTRATOR_TYPE=gke
+INSTANCE_COUNT=1
+NETWORK_CONFIG=default_network
+GPU_COUNT=2
+GKE_NODE_POOL_COUNT=2
+GKE_NODE_COUNT=2
```
[the remark this answers is not in the document]
Comment (soumyapani) [reply examples/gkeenv.list]: fixed in new iteration.

File: scripts/_env_var_util.sh [hunk 4]
```diff
@@ -97,8 +97,8 @@ EOF
 
     # setting instance count
     if [[ -z "$INSTANCE_COUNT" ]]; then
-        echo "INSTANCE_COUNT environment variable not found. Default value is 1"
-        export INSTANCE_COUNT=1
+        echo "INSTANCE_COUNT environment variable not found. Default value is 0."
+        export INSTANCE_COUNT=0
```
[the remark this answers is not in the document]
Comment (soumyapani) [reply scripts/_env_var_util.sh]: done.

File: examples/gkeenv.list (hunk 2 above)
[the remark this answers is not in the document]
Comment (soumyapani) [reply examples/gkeenv.list]: done.

**Review by DmitryKakurin — APPROVED** [review]

**Pull request merged**